enhancements/installer/aws-internal-clusters.md

```diff
@@ -0,0 +1,192 @@
+---
+title: (AWS) Internal OpenShift clusters
+authors:
+  - "@abhinavdahiya"
+reviewers:
+  - "@wking"
+  - "@sdodson"
+approvers:
+  - "@sdodson"
+creation-date: 2019-11-08
+last-updated: 2019-11-08
+status: implemented
+superseded-by:
+  - "https://docs.google.com/document/d/1IUCqz6AhxNwYDyx9jPcqbsa1-EloWD13OIONK7YP3JE"
+---
+
+# aws-internal-openshift-clusters
+
+## Release Signoff Checklist
+
+- [ x ] Enhancement is `implementable`
+- [ x ] Design details are appropriately documented from clear requirements
+- [ x ] Test plan is defined
+- [ x ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+Many customer environments don't require external connectivity from the outside world and as such, they would prefer not to expose the cluster endpoints to public. Currently the OpenShift installer exposes endpoints of the cluster like Kubernetes API, or OpenShift Ingress to the Internet, although most of these endpoints can be made Internal after installation with varying degree of ease individually, creating OpenShift clusters which are Internal by default is highly desirable for users.
+
+## Motivation
+
+### Goals
+
+Install an OpenShift cluster on AWS as internal/private, which is only accessible from my internal network and not visible to the Internet.
+
+### Non-Goals
+
+No additional isolation from other clusters in the network from the ones provided by [shared networking][aws-shared-networking]
+
+## Proposal
+
+To create Internal clusters, the installer binary needs access to the VPC where the cluster will be created to communicate with the cluster’s Kubernetes API, therefore, installing to [existing subnets][aws-shared-networking] will be a requirement. In addition to the network connectivity to the endpoints of the cluster and the installer binary should also be able to resolve the newly created DNS records for the cluster.
+
+No public subnets are required, since no public Load Balancer will be created. And, since no public records will be required, the current requirement for the existence of a public R53 Zone matching the `BaseDomain` will no longer be required. The installer still creates the private R53 Zone for the cluster.
```

Comment:
nit: "Load Balancer" -> "load balancers".  There's the Kube API and `*.apps`, and also any customer-requested public LoadBalancer Services.

Reply:
nit:

> And, since no public records will be required, the current requirement for the existence of a public R53 Zone matching the `BaseDomain` will no longer be required.

->

> And, since public records will not be needed, the requirement for a public Route 53 Zone matching the `BaseDomain` will be relaxed.

or some such.

Reply:
done https://github.com/openshift/enhancements/compare/d1df0a7526e3ead64c786034f1f50e29d556a5aa..d5621ab1053e0b04209359ebcd82d867ab38ac2c